app/Http/Controllers/UserController.php

```diff
@@ -1078,43 +1079,71 @@ public function socialCallback(Request $request, Validate $validate)
             }
             $user_email = $userSocial->getEmail();
             $social_name = $userSocial->getName();
-            $user = User::where(['email' => $user_email])->first();
-            if(empty($user)){
-                $splitName = Util::split_name($social_name);
-                $user = User::create([
-                    'first_name'    => $splitName[0],
-                    'last_name'     => $splitName[1],
-                    'email'         => $user_email
-                ]);
-                SocialUser::create([
-                    'user_id'       => $user->id,
-                    'social_email'  => $user_email,
-                    'provider_id'   => $userSocial->getId(),
-                    'provider'      => $provider,
-                    'social_name'   => $social_name,
-                ]);
-                $nickname = $user->first_name.'-'.$user->last_name;
-                $this->createNickname($user->id, $nickname);
-            }
-            $postUrl = URL::to('/') . '/oauth/token';
-            $payload = [
-                'grant_type' => 'password',
-                'client_id' => $request->client_id,
-                'client_secret' => $request->client_secret,
-                'username' => $user->email,
-                'password' => env('PASSPORT_MASTER_PASSWORD'),
-                'scope' => '*',
-            ];
-            $generateToken = Util::httpPost($postUrl, $payload);
-            if($generateToken->status_code == 200){
-                $data = [
-                    "auth" => $generateToken->data,
-                    "user" => new UserResource($user),
-                ];
-                $status = 200;
-                $message = trans('message.success.success');
+            $social_user = SocialUser::where(['social_email' => $user_email, 'provider' => $provider])->first();
+            if($request->user()){
```

Comment:
Can you please move multiple if else in a separate function or may be further break it down. As it is not very readable.